language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -308,23 +309,26 @@ impl LibraVM {
         &self,
         remote_cache: &StateViewCache<'_>,
         writeset_payload: &WriteSetPayload,
+        txn_sender: Option<AccountAddress>,
     ) -> Result<ChangeSet, Result<(VMStatus, TransactionOutput), VMStatus>> {
         let gas_schedule = zero_cost_schedule();
         let mut cost_strategy = CostStrategy::system(&gas_schedule, GasUnits::new(0));
 
         Ok(match writeset_payload {
             WriteSetPayload::Direct(change_set) => change_set.clone(),
-            WriteSetPayload::Script {
-                script,
-                execute_as: signer,
-            } => {
+            WriteSetPayload::Script { script, execute_as } => {
                 let mut tmp_session = self.0.new_session(remote_cache);
+                let args = convert_txn_args(script.args());
+                let senders = match txn_sender {
+                    Some(sender) => vec![*execute_as, sender],
+                    None => vec![*execute_as, *execute_as],
```

Comment:
why do we need to double up here?

Reply:
i.e. why not just `vec![*execute_as]`?

Reply:
I went for the double-up since it seems odd that some admin scripts would take one signer, and some would take two...but I like this idea that you had. Let me try it out :)